src/agent/communicator/src/http_client.cpp

```diff
@@ -241,6 +241,27 @@ namespace http_client
 
         if (res.result() != boost::beast::http::status::ok)
         {
+            if (res.result() == boost::beast::http::status::unauthorized ||
+                res.result() == boost::beast::http::status::forbidden)
+            {
+                std::string message {};
+
+                try
+                {
+                    message = nlohmann::json::parse(boost::beast::buffers_to_string(res.body().data()))
+                                  .at("message")
+                                  .get_ref<const std::string&>();
+                }
+                catch (const std::exception& e)
+                {
+                    LogError("Error parsing message in response: {}.", e.what());
+                }
+
+                if (message == "Invalid key" || message == "Agent does not exist")
+                {
+                    throw std::runtime_error(message);
```

Comment:
we won't throw if the response code is unauthorized or forbidden and if the message does not match. Would that be OK? because this looks like it is very coupled to the specific string we will find in the message.

Reply:
These are the messages that the server uses to indicate that the agent has not been recognized. Since this cannot be guessed by just analyzing the error code, we need to verify the message. cc @wazuh/devel-pyserver